recipe/build.sh

```diff
@@ -5,10 +5,18 @@ rm -rf build || true
 
 
 if [ ${cuda_compiler_version} != "None" ]; then
-  CUDA_CMAKE_OPTIONS="-DCMAKE_CUDA_COMPILER=${CUDA_HOME}/bin/nvcc -DCMAKE_CUDA_HOST_COMPILER=${CXX} -DCUDA_ARCH_NAME=All "
+  CUDA_CMAKE_OPTIONS="-DCUDA_ARCH_NAME=All"
```

Comment:
CMake says `CMAKE_CUDA_COMPILER` is unused [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=874072&view=logs&j=bb1c2637-64c6-57bd-9ea6-93823b2df951&t=fb54090b-f7dc-5308-f4ca-2446cf26a524 ):

```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CUDA_COMPILER
    CUDAToolkit_ROOT
    Python_EXECUTABLE
```

So went ahead and dropped it

Also we specify the host compiler to use on [CUDA 11]( https://github.com/conda-forge/nvcc-feedstock/blob/c01599568a71b44a3026970d6b5666f9c7e0e8be/recipe/linux/nvcc.sh#L9 ) and [CUDA 12]( https://github.com/conda-forge/cuda-nvcc-feedstock/blob/bb6f112003bc990193e31788b68d222f32e3129d/recipe/activate.sh#L41 ). So there is no need to specify it here